File: src/com/dafttech/terra/engine/gui/modules/GUIModule.java
package com.dafttech.terra.engine.gui.modules;

import com.dafttech.terra.engine.gui.containers.GUIContainer;

public abstract class GUIModule {
    public GUIContainer container;
    
    public GUIContainer getContainer() {
        return container;
    }
    
    public abstract void create();
    
    public void update(float delta) {
        
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/modules/GUIModule.scala
package com.dafttech.terra.engine.gui.modules

import com.dafttech.terra.engine.gui.containers.GUIContainer

abstract class GUIModule {
  var container: GUIContainer = null

  def getContainer: GUIContainer = container

  def create(): Unit

  def update(delta: Float): Unit = ()
}